dns/ddclient/src/opnsense/scripts/ddclient/lib/account/hetzner.py

```diff
@@ -144,34 +144,26 @@ def _get_record(self, headers, zone_id, record_name, record_type):
             return None
 
     def _update_record(self, headers, zone_id, record_name, record_type, address):
-        """Update existing record with new address"""
-        url = f"{self._api_base}/zones/{zone_id}/rrsets/{record_name}/{record_type}"
-
-        data = {
-            'records': [{'value': str(address)}],
-            'ttl': int(self.settings.get('ttl', 300))
-        }
+        """Update existing record with new address
 
-        response = requests.put(url, headers=headers, json=data)
+        NOTE: Hetzner Cloud API has a bug where PUT returns 200 but doesn't update.
+        Workaround: DELETE old record, then POST new record.
+        """
```

Comment:
Updating records cannot be done with `PUT /.../rrsets/...`, you must use one of the following endpoint:
- https://docs.hetzner.cloud/reference/cloud#zone-rrset-actions-set-records-of-an-rrset
- https://docs.hetzner.cloud/reference/cloud#zone-rrset-actions-add-records-to-an-rrset
- https://docs.hetzner.cloud/reference/cloud#zone-rrset-actions-remove-records-from-an-rrset

Also note, that those endpoint return "Actions", which describe async tasks that should be waited upon.